include/eve/module/real/proba/function/regular/generic/exponential.hpp

```diff
@@ -0,0 +1,223 @@
+
+    T lambda;
+  };
+
+
+  namespace detail
+  {
+    //////////////////////////////////////////////////////
+    /// cdf
+    template<floating_value T, floating_value U>
+    EVE_FORCEINLINE  auto cdf_(EVE_SUPPORTS(cpu_)
+                              , exponential<T> const &expo
+                              , U const &x ) noexcept
+    requires compatible_values<T, U>
+    {
+      return arithmetic_call(cdf, expo, x);
+    }
+
+    template<floating_value T>
+    EVE_FORCEINLINE  auto cdf_(EVE_SUPPORTS(cpu_)
+                              , exponential<T> const &expo
+                              , T const &x ) noexcept
+    {
+      return -expm1(if_else(is_ltz(x), zero, -x*expo.lambda));
+    }
+
+    //////////////////////////////////////////////////////
+    /// pdf
+    template<floating_value T, floating_value U>
+    EVE_FORCEINLINE  auto pdf_(EVE_SUPPORTS(cpu_)
+                              , exponential<T> const &expo
+                              , U const &x ) noexcept
+    requires compatible_values<T, U>
+    {
+      return arithmetic_call(pdf, expo, x);
+    }
+
+    template<floating_value T>
+    EVE_FORCEINLINE  auto pdf_(EVE_SUPPORTS(cpu_)
+                              , exponential<T> const &expo
+                              , T const &x ) noexcept
+    {
+      return expo.lambda*exp(-expo.lambda*x);
+    }
+
+    //////////////////////////////////////////////////////
+    /// invcdf
+    template<floating_value T, floating_value U>
+    EVE_FORCEINLINE  auto invcdf_(EVE_SUPPORTS(cpu_)
+                                 , exponential<T> const &expo
+                                 , U const &x ) noexcept
+    requires compatible_values<T, U>
+    {
+      return arithmetic_call(invcdf, expo, x);
+    }
+
+    template<floating_value T>
+    EVE_FORCEINLINE  auto invcdf_(EVE_SUPPORTS(cpu_)
+                                 , exponential<T> const &expo
+                                 , T const &p ) noexcept
+    {
+      return -log1p(-p)/expo.lambda;
+    }
+
+    //////////////////////////////////////////////////////
+    /// mgf
+    template<floating_value T, floating_value U>
+    EVE_FORCEINLINE  auto mgf_(EVE_SUPPORTS(cpu_)
+                              , exponential<T> const &expo
+                              , U const &x ) noexcept
+    requires compatible_values<T, U>
+    {
+      return arithmetic_call(mgf, expo, x);
+    }
+
+    template<floating_value T>
+    EVE_FORCEINLINE  auto mgf_(EVE_SUPPORTS(cpu_)
+                              , exponential<T> const &expo
+                              , T const &t ) noexcept
+    {
+      return if_else(t < expo.lambda, expo.lambda/(expo.lambda-t),  allbits);
+    }
+
+    //////////////////////////////////////////////////////
+    /// parameters
+    template<floating_value T>
+    EVE_FORCEINLINE  auto parameters_(EVE_SUPPORTS(cpu_)
+                                     , exponential<T> const &expo) noexcept
+    {
+      return expo.lambda; 
```

Comment:
Same, we should just return the structure and let structured binding do its job.
There we have one case returning a tuple, one not. This is quite confusing.